metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceImpl.java

```diff
@@ -260,11 +281,23 @@ public void notifyOfTableUpdate(@Nonnull final MetacatUpdateTablePostEvent event
             this.publishNotification(this.tableTopicArn, message);
             CounterWrapper.incrementCounter("metacat.notifications.sns.tables.update.succeeded");
         } catch (final Exception e) {
-            log.error("Unable to publish update table notification", e);
-            CounterWrapper.incrementCounter("metacat.notifications.sns.tables.update.failed");
+            handleException(event.getName(), "Unable to publish update table notification",
+                "metacat.notifications.sns.tables.update.failed", message, e);
         }
     }
 
+    private void handleException(final QualifiedName name, final String message, final String counterKey,
+        final SNSMessage payload, final Exception e) {
+        log.error(String.format("%s with payload: %s", message, payload), e);
+        DynamicCounter.increment(counterKey, getQualifiedNameTagList(name));
+    }
+
+    private static BasicTagList getQualifiedNameTagList(final QualifiedName name) {
```

Comment:
This seems like a utility method that should be more global no? Also why is it static?